yapapi/golem.py

```diff
@@ -117,13 +128,31 @@ def __init__(
 
     @property
     def driver(self) -> str:
-        """Name of the payment driver"""
-        return self._engine.driver
+        """Name of the payment driver.
+
+        This property is deprecated, please use `payment_driver` instead.
+        """
+        warn_deprecated("driver", "payment_driver", "0.7.0", Deprecated.property)
```

Comment:
hmm @zakaprov is there a strong reason why we're using `warn_deprecated` function instead of the `@deprecated` decorator here?

Reply:
@shadeofblue Yes, `deprecated` does not support properties. The library raises a `TypeError` when trying to do so.